### PR TITLE
`binary-dedupe` changes for Spark 4.0.0 [databricks]

### PR DESCRIPTION
**Pull request by razajafri**
This is to address the fact that the name of this shim is not `spark3*` but `spark4*`

We also renamed the `spark-common` to `spark-shared` anywhere it was relevant 

contributes to #9259

### CODE REVIEW

File: dist/scripts/binary-dedupe.sh [hunk 1]
```diff
@@ -173,7 +173,7 @@ function verify_same_sha_for_unshimmed() {
   # TODO currently RapidsShuffleManager is "removed" from /spark3* by construction in
```
Comment (razajafri) [inline dist/scripts/binary-dedupe.sh:173]: This might need to be addressed for spark4, will take another look

Comment (razajafri) [reply dist/scripts/binary-dedupe.sh]: I have updated the comment

**Comment by razajafri** [comment]
build

**Comment by firestarman** [comment]
build

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

File: dist/maven-antrun/build-parallel-worlds.xml [hunk 2]
```diff
@@ -131,7 +131,7 @@
 
         <!-- Remove the explicily unshimmed files from the common directory -->
         <delete>
-            <fileset dir="${project.build.directory}/parallel-world/spark3xx-common"
+            <fileset dir="${project.build.directory}/parallel-world/spark34-common"
```
Comment (gerashegalov) [inline dist/maven-antrun/build-parallel-worlds.xml:134]: spark34 may confusingly suggest spark 3.4 for the reader

Let us just do spark-common or spark-shared. spark3xx was not an ideal choice. What if this code survives until Spark 5 :)

**Comment by razajafri** [comment]
@gerashegalov I have addressed your concern PTAL

**Comment by razajafri** [comment]
build

**Comment by gerashegalov** [comment]
Putting this PR into Draft mode to avoid merging. We should only merge it after #10994 without which it is hard to verify whether this PR is working

**Comment by razajafri** [comment]
> Putting this PR into Draft mode to avoid merging. We should only merge it after #10994 without which it is hard to verify whether this PR is working

#10994 will not be successful without this PR either unless we build without binary dedupe? All of the PRs for Spark 4.0 were merged by verifying manually by pulling changes from pom branch. Whats different about this PR?

**Comment by gerashegalov** [comment]
> #10994 will not be successful without this PR either unless we build without binary dedupe?

In this case 10994's description should say it depends on this PR. It would also make sense to combine them. Suppose we merge this on Monday morning, 10994 can be merged as well. If the nightly tests will uncover an issue and we need to revert this PR, then we have to revert 10994 as well

**Comment by razajafri** [comment]
> > #10994 will not be successful without this PR either unless we build without binary dedupe?
> 
> In this case 10994's description should say it depends on this PR. It would also make sense to combine them. Suppose we merge this on Monday morning, 10994 can be merged as well. If the nightly tests will uncover an issue and we need to revert this PR, then we have to revert 10994 as well

I can modify the #10994 description to say it depends on this

File: dist/scripts/binary-dedupe.sh [hunk 3]
```diff
@@ -34,10 +34,10 @@ case "$OSTYPE" in
 esac
 
 STEP=0
-export SPARK3XX_COMMON_TXT="$PWD/spark3xx-common.txt"
-export SPARK3XX_COMMON_COPY_LIST="$PWD/spark-common-copy-list.txt"
+export SPARK_COMMON_TXT="$PWD/spark-shared.txt"
+export SPARK_COMMON_COPY_LIST="$PWD/spark-common-copy-list.txt"
```
Comment (mythrocks) [inline dist/scripts/binary-dedupe.sh:38]: Sorry for nitpicking, but could we pick a lane here?  Is it `shared` or `common`?

Comment (razajafri) [reply dist/scripts/binary-dedupe.sh]: Done
PTAL

**Comment by razajafri** [comment]
build

**Review by mythrocks — APPROVED** [review]

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
premerge was successful but failed to report it's status here

**Review by gerashegalov — APPROVED** [review]
LGTM

File: dist/maven-antrun/build-parallel-worlds.xml [hunk 4]
```diff
@@ -131,7 +131,7 @@
 
         <!-- Remove the explicily unshimmed files from the common directory -->
         <delete>
-            <fileset dir="${project.build.directory}/parallel-world/spark3xx-common"
+            <fileset dir="${project.build.directory}/parallel-world/spark-shared"
```
Comment (jlowe) [inline dist/maven-antrun/build-parallel-worlds.xml:134]: There are a lot of references to spark3xx-common in the code base, not sure all of them have been updated to the new name, e.g.: coverage-report, shims.md, spark-premerge-build.sh, etc.  Don't we need to update those as well?

Comment (razajafri) [reply dist/maven-antrun/build-parallel-worlds.xml:134]: Thanks for pointing that out I will create a follow-on PR

Comment (razajafri) [reply dist/maven-antrun/build-parallel-worlds.xml:134]: https://github.com/NVIDIA/spark-rapids/pull/11066

**Pull request merged**